app/controllers/concerns/rendering.rb

```diff
@@ -8,17 +8,24 @@ module Base
       include ActionController::MimeResponds
 
       # overload render method to automatically set content type
-      def render(args, ...)
-        case args
-        in jsonapi:
+      def render(options, ...)
+        mime_type, * = Mime::Type.parse(response.content_type.to_s) rescue nil
+
+        # skip if we've already set content type
+        unless mime_type.nil?
+          return super
+        end
+
+        case options
+        in jsonapi: _
           response.content_type = Mime::Type.lookup_by_extension(:jsonapi)
-        in json:
+        in json: _
```

Comment:
This can respond with JSONAPI depending on route format and accept header, since JSON and JSONAPI are often synonymous.